src/Compilers/CSharp/Portable/CSharpCompilationOptions.cs

```diff
@@ -145,6 +105,7 @@ public sealed class CSharpCompilationOptions : CompilationOptions, IEquatable<CS
             ReportDiagnostic generalDiagnosticOption,
             int warningLevel,
             IEnumerable<KeyValuePair<string, ReportDiagnostic>> specificDiagnosticOptions,
+            IEnumerable<(SyntaxTree, IEnumerable<(string, ReportDiagnostic)>)> perTreeDiagnosticOptions,
```

Comment:
this tuple type is unclear.  Can you either give member names to it, or make an actual named type.